lib/services/nodes-api-service.js

```diff
@@ -509,6 +509,26 @@ function nodeApiServiceFactory(
     };
 
     /**
+     * Remove the tag from a list of nodes
+     * @param  {String}     tagName
+     * @return {Promise}
+     */
+    NodeApiService.prototype.masterDelTagById = function(tagName) {
+        return Promise.resolve().then(function() {
+                assert.string(tagName, 'tag must be a string');
```

Comment:
@uppalk1 This assert doesn't need to be in the Promise chain.  You could simplify this code a bit by moving the assert out of the promise chain and beginning the chain with `return waterline.nodes.findByTag(tagName).map(...)`